src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -455,9 +455,16 @@ def _special_cases(self, text, cmd, outside):
             if text[0] == SELECT_SYMBOL['outside']:
                 cmd = text[1:]
                 outside = True
-                if cmd.strip() and cmd.split()[0] == 'cd':
-                    self.handle_cd(parse_quotes(cmd))
-                    continue_flag = True
+                if cmd.strip():
+                    parsed_cmd = parse_quotes(cmd)
+                    if cmd.split()[0] == 'cd':
```

Comment:
split the string first then enter the if condition statement to avoid splitting the string multiple items.